clients/filesystem-fuse/src/filesystem_metadata.rs

```diff
@@ -0,0 +1,196 @@
+    file_stat_map: HashMap<u64, FileStat>,
+
+    // dir_name_map stores the inode of a directory by its full path of mount point.
+    dir_name_map: HashMap<String, u64>,
+
+    // dir_child_map stores the children of a directory.
+    dir_child_map: HashMap<u64, HashMap<String, u64>>,
+
+    // inode_id_generator is used to generate unique inode IDs.
+    inode_id_generator: u64,
+}
+
+impl DefaultFileSystemMetadata {
+    pub fn new() -> Self {
+        Self {
+            file_stat_map: Default::default(),
+            dir_name_map: Default::default(),
+            dir_child_map: Default::default(),
+            inode_id_generator: 1000000,
+        }
+    }
+
+    pub(crate) fn next_inode_id(&mut self) -> u64 {
+        self.inode_id_generator += 1;
+        self.inode_id_generator
+    }
+}
+
+impl IFileSystemMetadata for DefaultFileSystemMetadata {
+    fn get_file_path(&self, file_id: u64) -> String {
+        self.file_stat_map
+            .get(&file_id)
+            .map(|x| x.path.clone())
+            .unwrap_or_else(|| "".to_string())
+    }
+
+    fn add_file_with_name(&mut self, name: &str, parent: &str) -> FileStat {
+        let parent_inode = self.dir_name_map.get(parent).unwrap();
+        self.add_file(*parent_inode, name)
+    }
+
+    fn add_file(&mut self, parent_inode: u64, name: &str) -> FileStat {
+        let file_stat = FileStat::new_file(name, self.next_inode_id(), parent_inode);
+        self.file_stat_map
+            .insert(file_stat.inode, FileStat::clone(&file_stat));
+        self.dir_child_map
+            .entry(parent_inode)
+            .or_insert(HashMap::new())
+            .insert(name.to_string(), file_stat.inode);
+        file_stat
+    }
+
+    fn add_root_dir(&mut self) -> FileStat {
+        let file_stat = FileStat::new_dir("", 1, 0);
+        self.file_stat_map
+            .insert(file_stat.inode, FileStat::clone(&file_stat));
+        self.dir_child_map.insert(file_stat.inode, HashMap::new());
+
+        self.dir_name_map
+            .insert(file_stat.name.clone(), file_stat.inode);
+        file_stat
+    }
+
+    fn add_dir(&mut self, parent_inode: u64, name: &str) -> FileStat {
+        let file_stat = FileStat::new_dir(name, self.next_inode_id(), parent_inode);
+        self.file_stat_map
+            .insert(file_stat.inode, FileStat::clone(&file_stat));
+        self.dir_child_map.insert(file_stat.inode, HashMap::new());
+        self.dir_child_map
+            .get_mut(&parent_inode)
+            .unwrap()
+            .insert(name.to_string(), file_stat.inode);
+
+        let parent = self.file_stat_map.get(&parent_inode).unwrap().clone();
+        let full_name = format!("{}/{}", parent.path, name);
+        self.dir_name_map.insert(full_name, file_stat.inode);
+        file_stat
+    }
+
+    fn add_dir_with_name(&mut self, parent: &str, name: &str) -> FileStat {
+        let parent_inode = self.dir_name_map.get(parent).unwrap().clone();
```

Comment:
it's not recommended to `unwrap` directly for all codes

Reply:
Ok, I understand. I will only use unwrap when it is guaranteed to succeed.